packages/cli/src/ui/components/shared/RadioButtonSelect.tsx

```diff
@@ -71,30 +76,81 @@ export function RadioButtonSelect<T>({
     }
   }, [activeIndex, items.length, scrollOffset, maxItemsToShow]);
 
+  useEffect(
+    () => () => {
+      if (numberInputTimer.current) {
+        clearTimeout(numberInputTimer.current);
+      }
+    },
+    [],
+  );
+
   useInput(
     (input, key) => {
+      const isNumeric = showNumbers && /^[0-9]$/.test(input);
+
+      // Any key press that is not a digit should clear the number input buffer.
+      if (!isNumeric && numberInputTimer.current) {
```

Comment:
minor edge case: any thoughts on what should happen if the user presses ctrl-5 or similar. Should that be treated as a number or a non-number.

Reply:
`ctrl-5` wouldn't count as an entry, do we envision users using other key combinations with numbers for selecting?

Reply:
good. want to make sure only raw numbers show up for selection. particularly if we start letting users set arbitrary keyboard maps we need to make sure our core ui controls only trigger for limited keystrokes.